### PR TITLE
Fix get_re_matches error with deep paths

### LINKED ISSUE #781: IsADirectoryError: [Errno 21] Is a directory:
**Describe the bug**

When trying to run the latest version from develop (commit ad6260f79fc134292dd57bbb9cc395d7d9da1dc6), I'm getting the following unclear error:

```
semgrep -f ../../registry/.semgrep.yml ../../<> --exclude-dir=node_modules 
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/omerlh/dev/semgrep/semgrep/semgrep/core_runner.py", line 38, in get_re_matches
    contents = path.read_text()
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/pathlib.py", line 1216, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/pathlib.py", line 1203, in open
    opener=self._opener)
IsADirectoryError: [Errno 21] Is a directory: '<>'
"""
```

**Environment**
If not using semgrep.live: are you running off docker, an official binary, a local build?

### PR DESCRIPTION
**Pull request by underyx**
Closes https://github.com/returntocorp/semgrep/issues/781

### CODE REVIEW

File: semgrep/semgrep/core_runner.py [hunk 1]
```diff
@@ -35,7 +35,11 @@
 
 
 def get_re_matches(patterns_re: List[Tuple], path: Path) -> List[PatternMatch]:
-    contents = path.read_text()
+    try:
+        contents = path.read_text()
+    except UnicodeDecodeError:
```
Comment (mschwager) [inline semgrep/semgrep/core_runner.py:40]: Do we want to be more general than `UnicodeDecodeError` here, e.g. just catch `Exception` and warn the user something went wrong? I took a look at the `read_text` docs: https://docs.python.org/3/library/pathlib.html?highlight=path#pathlib.Path.read_text and it looks like the `errors` kwarg can be utilized too. It looks like `errors` defaults to `strict` (makes sense here, we can't easily regex non-text contents), which [will raise a `ValueError`](https://docs.python.org/3/library/functions.html#open) if there's an encoding error. So here we got `ValueError` -> [`UnicodeError`](https://docs.python.org/3/library/exceptions.html#UnicodeError) -> `UnicodeDecodeError`.

Generally, though, it's better to fail quick and loud and learn about cases we didn't consider. I'd say we should catch `UnicodeDecodeError` here, QA this functionality over a large corpus, look for failures, and patch this up with our learnings. Thoughts?

**Comment by mschwager** [comment]
Can we just tackle #785 here too?

File: semgrep/semgrep/core_runner.py (hunk 1 above)
Comment (underyx) [reply semgrep/semgrep/core_runner.py:40]: I'd leave it like this before #785 is done, and like you said, find more bugs to fix that way that would get their handling filed as new issues.

**Comment by underyx** [comment]
> Can we just tackle #785 here too?

I'm not against it being added to this PR by someone else, but I shouldn't spend much more time on this PR and I haven't yet looked at how `pattern-regex` works at all so I don't wanna commit to doing it blindly.

**Comment by brendongo** [comment]
I'll do #785 in a separate PR

**Review by brendongo — APPROVED** [review]

**Pull request merged**